### DISCUSSION TITLE
Not possible to await getters variable

### DISCUSSION THREAD

**Discussion by ayophanz** [vuejs/pinia#1613] [Help and Questions] [answered]
### Reproduction

none

### Steps to reproduce the bug

I want to await for pinia getters variable to before router execute, here example (image).
middleware
![image](https://user-images.githubusercontent.com/24581584/187582603-1288ee86-4cac-44a4-8178-45f7f473fab4.png)
store
![image](https://user-images.githubusercontent.com/24581584/187582679-7f9e0fd4-ee4e-407b-8c6f-a822a2c72726.png)

### Expected behavior

It should work similar to service await
here
![image](https://user-images.githubusercontent.com/24581584/187582907-70a5bc59-dd5e-49ca-91eb-9b6a3c8aee3c.png)

### Actual behavior

I have login page, when I try to login it will not await the pinia getters variable to access the value first, as a result I'm failing tp login.

**Comment by dbuckman** [comment]
I think you need an async declaration on the function.  Something like `async getIsAuthenticated()`

**Reply by posva** [reply, marked as answer]
The async seems unnecessary. Unfortunately this is just too old and the op never provided enough info
